packages/components/src/components/button-toggle/button-highlight-wrapper.jsx

```diff
@@ -3,77 +3,66 @@ import PropTypes from 'prop-types';
         }
-    }
+        return () => resetHighlight();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    componentWillUnMount() {
-        this.resetHighlight();
-    }
+    React.useEffect(() => {
+        const active_button_el = [...wrapper_ref.current?.getElementsByClassName(class_selector)][0];
+        if (active_button_el) updateHighlightPosition(active_button_el);
+        else if (left !== 0) resetHighlight(); // clear highlight when active element doesn't exist
+    });
 
-    onClick = (e, buttonClick) => {
+    const onClick = (e, buttonClick) => {
         if (!e.target) return;
-        this.updateHighlightPosition(e.target.closest('button'));
+        updateHighlightPosition(e.target.closest('button'));
         if (typeof buttonClick === 'function') {
             buttonClick();
         }
     };
 
-    resetHighlight = () => {
-        this.setState({ left: 0 });
-    };
+    const resetHighlight = () => setLeft(0);
 
-    updateHighlightPosition = el => {
+    const updateHighlightPosition = el => {
         if (!el) return;
-        const { offsetLeft: left } = el;
-        if (this.state.left !== left) {
-            this.setState({ left });
-        }
+        const { offsetLeft } = el;
+        if (left !== offsetLeft) setLeft(offsetLeft);
     };
 
-    render() {
-        const { children, className, has_rounded_button, ...other_props } = this.props;
-        const props = {
-            className: classnames('dc-button-menu__wrapper', className),
-            ...other_props,
-        };
-        const button_width = (100 / children.length).toFixed(2);
+    const props = {
+        className: classnames('dc-button-menu__wrapper', className),
+        ...other_props,
+    };
+    const button_width = (100 / children.length).toFixed(2);
 
-        return (
-            <div ref={node => (this.node = node)} {...props}>
-                {React.Children.map(children, child =>
-                    React.cloneElement(child, {
-                        onClick: e => this.onClick(e, child.props.onClick),
-                    })
-                )}
-                <Highlight left={this.state.left} width={`${button_width}%`} has_rounded_button={has_rounded_button} />
-            </div>
-        );
-    }
-}
+    return (
+        <div ref={wrapper_ref} {...props}>
+            {React.Children.map(children, child =>
+                React.cloneElement(child, {
+                    onClick: e => onClick(e, child.props.onClick),
+                })
+            )}
+            <Highlight left={left} width={`${button_width}%`} has_rounded_button={has_rounded_button} />
+        </div>
+    );
+};
 
 HighlightWrapper.propTypes = {
     children: PropTypes.array,
     className: PropTypes.string,
     has_rounded_button: PropTypes.bool,
-    timeout: PropTypes.number,
 };
 
-export default HighlightWrapper;
+export default React.memo(HighlightWrapper);
```

Comment:
This doesn't work unless we memoize all other use-cases. [for more information](https://gist.github.com/slikts/fd3768de1493419ed9506002b452fcdc#a-flat-list-of-primitive-values-is-the-ideal-case-for-dependencies-in-react)

Is it possible to throw a warning or something anytime the children is not memoized? Otherwise I think this would fall under premature optimization.

Reply:
Thank you for this nice catch! @reza-fs 
As this component is used only once we can simply pass memoized children to it. I've done this. Please check and let me know if this resolves the issue :pray: